src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs

```diff
@@ -0,0 +1,131 @@
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+    }
+}
+";
+
+            var expected = @"using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine();
+    }
+}
+";
+            return AssertCodeCleanupResult(expected, code,
+                (CodeCleanupOptions.AreCodeCleanupRulesConfigured, enabled: true),
+                (CodeCleanupOptions.RemoveUnusedImports, enabled: true));
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public Task FormatDocumentSortUsings()
+        {
+            var code = @"using System.Collections.Generic;
+using System;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var list = new List<int>();
+        Console.WriteLine(list.Count);
+    }
+}
+";
+
+            var expected = @"using System;
+using System.Collections.Generic;
+class Program
+{
+    static void Main(string[] args)
+    {
+        var list = new List<int>();
+        Console.WriteLine(list.Count);
+    }
+}
+";
+            return AssertCodeCleanupResult(expected, code,
+                (CodeCleanupOptions.AreCodeCleanupRulesConfigured, enabled: true),
+                (CodeCleanupOptions.SortImports, enabled: true));
+        }
+
+        [Fact(Skip = "disable the test temporarily until figure out how to set up diagnostic analyzer")]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public Task FormatDocumentRemoveUnusedVariable()
+        {
+            var code = @"class Program
+{
+    void Method()
+    {
+        int a;
+    }
+}
+";
+            var expected = @"class Program
+{
+    void Method()
+    {
+    }
+}
+";
+            return AssertCodeCleanupResult(expected, code,
+                (CodeCleanupOptions.AreCodeCleanupRulesConfigured, enabled: true),
+                (CodeCleanupOptions.RemoveUnusedVariables, enabled: true));
+
+            //workspace.Options = workspace.Options.WithChangedOption(RemoteFeatureOptions.DiagnosticsEnabled, false);
+        }
+
+        protected static async Task AssertCodeCleanupResult(string expected, string code, params (PerLanguageOption<bool> option, bool enabled)[] options)
+        {
+            using (var workspace = TestWorkspace.CreateCSharp(code))
+            {
+                if (options != null)
+                {
+                    foreach (var option in options)
+                    {
+                        workspace.Options = workspace.Options.WithChangedOption(option.option, LanguageNames.CSharp, option.enabled);
+                    }
+                }
+
+
+                var hostdoc = workspace.Documents.Single();
+                var document = workspace.CurrentSolution.GetDocument(hostdoc.Id);
+
+                var codeCleanupService = document.GetLanguageService<ICodeCleanupService>();
+                var newDoc = await codeCleanupService.CleanupAndFormatDocumentAsync(document, new CancellationToken());
```

Comment:
CancellationToken.None

Reply:
I think you don't need to call out Format,. I think CleanupAsync(Document document, cancellationToken) would be fine.

Reply:
fixed